common_mips.h

```diff
@@ -43,6 +43,7 @@ USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #ifndef ASSEMBLER
 
+#if !defined(MIPS24K)
```

Comment:
@brada4 Ok now it compiles fine also  without special flags. I suppose that this `if` removed assembly but I see:

```
static inline unsigned int rpcc(void){
  unsigned long ret;

  __asm__ __volatile__(".set   push    \n"
          "rdhwr %0, $30  \n"
          ".set pop" : "=r"(ret) : : "memory");

  return ret;
}
```
So seems is working on this PR. I don't know why it doesn't work on master.

Reply:
Ok I suppose that the negation is wrong. But it is working :smile:

Reply:
On master (or rather, develop branch) we have only the outer "ifndef ASSEMBLER" - meaning "when this is included from C code like blas_server.c" - the PR adds the "and we are not building for the 24K" condition so in your case this rpcc function is not included (and RPCC_DEFINED is not set, so the other code "knows" that it has to work without it)

Reply:
ok

Reply:
@martin-frbg So is it ok if I see this resolved as

```
cat ./build_dir/target-mipsel_24kc_musl/OpenBLAS-0.3.9/common_mips.h
#ifndef ASSEMBLER

static inline unsigned int rpcc(void){
  unsigned long ret;

  __asm__ __volatile__(".set   push    \n"
          "rdhwr %0, $30  \n"
          ".set pop" : "=r"(ret) : : "memory");

  return ret;
}
#define RPCC_DEFINED
```

Reply:
Sorry, I do not understand your question. What you have shown here looks like the "old" code from develop before this PR.

Reply:
Yes sorry with all these building dirs on single commits I took an old one.

Reply:
Guess it is time to commit this to avoid further misunderstanding ?